lib/dotinstall/util.py

```diff
@@ -115,6 +115,31 @@ def create_symlink(source: str, target: str) -> None:
     os.symlink(source, target)
 
 
+def concat_files(sources, target, comment_prefix=None):
+    for i, _ in enumerate(sources):
+        sources[i] = os.path.abspath(os.path.expanduser(sources[i]))
+    target = os.path.expanduser(target)
+    _remove_symlink(target)
+    create_dir(os.path.dirname(target))
+    print('Concatenating %s to %s...' % (sources, target))
+    with open(target, 'w') as ft:
+        if comment_prefix:
+            ft.write('%sDO NOT EDIT THIS FILE%s'
+                     % (comment_prefix, os.linesep))
+            ft.write('%sThis file was concatenated by dotfiles utility.%s'
+                     % (comment_prefix, os.linesep))
+            ft.write('%sAll changes in this file will not be retained.%s%s'
+                     % (comment_prefix, os.linesep, os.linesep))
+        for source in sources:
+            with open(source) as fs:
+                if comment_prefix:
+                    ft.write('%sConcatenated file %s of %s from: %s%s%s'
+                             % (comment_prefix, sources.index(source)+1,
+                                len(sources), source, os.linesep, os.linesep))
+                ft.write(fs.read())
+                ft.write(os.linesep)
```

Comment:
Let's change the logic a little bit and make sure the concatenated files do not end with a double line separator (e.g. one blank line):

```python3
for i, source in enumerate(sources):
    if i > 0:
        print(file=handle)
    print(fs.read(), file=handle)
```

also to ensure better scope let's make sure the source path is opened only for the read operation, not for the whole write block

Reply:
Opted for `pathlib.Path` approach. Resolving.

Reply:
The first part of this remark is still to be resolved

Reply:
Oh right, done.